src/loader/loader.js

```diff
@@ -20,7 +20,7 @@ export function loader(
         };
 
         // The AWS RUM client's JavaScript API
-        window[n] = function (c, p) {
+        window[n] = function (c, ...p) {
```

Comment:
issue: I'm not sure it is safe for us to use a spread operator here. This is the code snippet which is copied into the application. Specifically:
1. It is not backwards compatible. In the case where customers update the web client version, but not the snippet, the Evidently features will not work.
2. The spread operator is not supported by ES5, which the snippet must use.

Reply:
1. You're right, if the user updates the web client but not the snippet they can't use the Evidently feature, but the Evidently feature is new and if they want to use it we're just telling them they have to update in both places. I think it'd only be an issue if it broke old features, but updating just one of the web client / the snippet won't break anything if the customer continues to use the web client in the same way.
2. What JS compilers are we trying to support? [Spread syntax has been implemented by every major JS runner for at least the last 6 years](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#specifications).

Reply:
What is the rational for breaking convention and using this signature
```
cwr('getEvaluations', ['featureA', 'featureB'], (err, data) -> { ... });
```
instead of this signature?
```
cwr('getEvaluations', { features: ['featureA', 'featureB'], callback: (err, data) -> { ... } });
```

Regarding (2), it's about what browsers do RUM users need to support? I don't think we should make assumptions about that, so the safest thing to do is to use the lowest common denominator, which is ES5.

Reply:
The original rationale was to separate the callback and the request so that it's easier to translate between CDN and SDK syntax and to avoid adding a new data structure just for the callback request.

After thinking about it more, you're correct in that if the user only updates one of the script/webclient then it'll break, to support this you're right that we'll have to switch to just taking a single object.

Right now it's implemented as:
```
cwr('evaluateFeature', { feature: 'featureA', callback: (err, data) -> { ... } });
const results: Promise<EvaluationResult> = AwsRum.evaluateFeature(feature: string);
```

An alternative would be:
```
cwr('evaluateFeature', { request: { feature: 'featureA' }, callback: (err, data) -> { ... } });
const results: Promise<EvaluationResult> = AwsRum.evaluateFeature(request: { feature: string });
```

The alternative is potentially clearer for translating between sdk and cdk syntax and would allow us to add additional parameters in the future, but could also just be adding unnecessary overhead if we don't think we're likely to want to change the request structure.

Reply:
Gotcha. I agree - the alternative you're proposing lgtm.

BTW I like what you've done with returning a promise from Orchestration. I think that's a pattern we can adopt for other things going forward.